program-libs/hasher/src/hash_to_field_size.rs

```diff
@@ -1,46 +1,23 @@
 use arrayvec::ArrayVec;
+use borsh::BorshSerialize;
 
-use crate::{keccak::Keccak, to_byte_array::ToByteArray, Hasher, HasherError};
+use crate::{keccak::Keccak, Hasher, HasherError};
 
 pub const HASH_TO_FIELD_SIZE_SEED: u8 = u8::MAX;
 
 pub trait HashToFieldSize {
     fn hash_to_field_size(&self) -> Result<[u8; 32], HasherError>;
 }
 
-impl<const N: usize> HashToFieldSize for [u8; N] {
-    fn hash_to_field_size(&self) -> Result<[u8; 32], HasherError> {
-        Ok(hash_to_bn254_field_size_be(self.as_slice()))
-    }
-}
-
-impl HashToFieldSize for String {
-    fn hash_to_field_size(&self) -> Result<[u8; 32], HasherError> {
-        Ok(hash_to_bn254_field_size_be(self.as_bytes()))
-    }
-}
-
-#[cfg(not(feature = "pinocchio"))]
-impl HashToFieldSize for crate::Pubkey {
-    fn hash_to_field_size(&self) -> Result<[u8; 32], HasherError> {
-        Ok(hash_to_bn254_field_size_be(self.as_ref()))
-    }
-}
-
-impl<T> HashToFieldSize for Vec<T>
+impl<T> HashToFieldSize for T
 where
-    T: ToByteArray,
+    T: BorshSerialize,
 {
     fn hash_to_field_size(&self) -> Result<[u8; 32], HasherError> {
-        let mut arrays = Vec::with_capacity(self.len());
-        for item in self {
-            let byte_array = item.to_byte_array()?;
-            arrays.push(byte_array);
-        }
-        let mut slices = Vec::with_capacity(self.len() + 1);
-        arrays.iter().for_each(|x| slices.push(x.as_slice()));
+        let borsh_vec = self.try_to_vec().map_err(|_| HasherError::BorshError)?;
         let bump_seed = [HASH_TO_FIELD_SIZE_SEED];
-        slices.push(bump_seed.as_slice());
+        let mut slices = [&[], bump_seed.as_slice()];
+        slices[0] = borsh_vec.as_slice();
```

Comment:
cant we just do this and remove the mut?
```suggestion
        let slices = [borsh_vec.as_slice(), bump_seed.as_slice()];
```